openedx/features/badging/helpers.py

```diff
@@ -1,52 +1,90 @@
+import logging
+
 from collections import OrderedDict
 
+from lms.djangoapps.courseware.courses import get_course_by_id
+from lms.djangoapps.teams import is_feature_enabled as is_teams_feature_enabled
+from lms.djangoapps.teams.models import CourseTeam
+from nodebb.constants import TEAM_PLAYER_ENTRY_INDEX
+from nodebb.models import TeamGroupChat
+
+from .constants import (
+    BADGES_KEY,
+    COURSE_ERROR,
+    FILTER_BADGES_ERROR,
+    TEAM_PLAYER
+)
 from .models import Badge
 
+log = logging.getLogger('edx.badging')
+
 
-def populate_trophycase(courses, earned_badges):
+def populate_trophycase(user, courses, earned_badges):
     """
     This method populate trophycase data into dictionary and returns it. This method has a hierarchy of
     enrolled courses, badge type, earned and yet to be earned badges with complete detail
     :param courses: Courses enrolled by user
     :param earned_badges: Badges earned by user
+    :param user: Current logged-in user
     :return: dictionary containing trophycase json
     """
     trophycase_dict = OrderedDict()
     badge_queryset = Badge.objects.all().order_by('threshold')
 
     for course_id, display_name in courses:
-
-        course_badges = get_all_badges(course_id, earned_badges, badge_queryset)
+        course_badges = get_course_badges(user, course_id, earned_badges, badge_queryset)
 
         trophycase_dict[unicode(course_id)] = {
             'display_name': display_name,
-            'badges': course_badges
         }
 
+        trophycase_dict[unicode(course_id)].update(course_badges)
+
     return trophycase_dict
 
 
-def get_all_badges(course_id, earned_badges, badge_queryset=None):
+def get_course_badges(user, course_id, earned_badges, badge_queryset=None):
     """
     Get all badges of a course in a hierarchy, categorised by badge type
+    :param user: Current logged-in user
     :param badge_queryset: Badge queryset
     :param course_id: Course identifier
     :param earned_badges: All badges earned in a course
     :return: List of badges in a course
     """
-    badges = list()
+    badges = {
+        BADGES_KEY: dict()
+    }
 
     if not badge_queryset:
         badge_queryset = Badge.objects.all().order_by('threshold')
 
     for badge_type, _ in Badge.BADGE_TYPES:
+
+        if badge_type == TEAM_PLAYER[TEAM_PLAYER_ENTRY_INDEX]:
+            course = get_course_by_id(course_id)
+
+            if not course:
+                error = COURSE_ERROR.format(course_id=course_id)
+                log.exception(error)
+                raise Exception(error)
```

Comment:
Since `course_id` is being passed to the function and does not change every loop, let's move this logic at the very top the function.

Reply:
The check is on `course` which is only required for team badges, hence cannot move this logic